sklearn/cross_validation.py

```diff
@@ -1042,14 +1042,22 @@ def cross_val_predict(estimator, X, y=None, cv=None, n_jobs=1,
                                                       train, test, verbose,
                                                       fit_params)
                             for train, test in cv)
-    p = np.concatenate([p for p, _ in preds_blocks])
+
+    p = [p for p, _ in preds_blocks]
     locs = np.concatenate([loc for _, loc in preds_blocks])
+
     if not _check_is_partition(locs, _num_samples(X)):
         raise ValueError('cross_val_predict only works for partitions')
+    inv_locs = np.empty(len(locs), dtype=int)
+    inv_locs[locs] = np.arange(len(locs))
+
+    # Check for sparse predictions
+    if sp.issparse(p[0]) :
+        p = sp.vstack(p, format = 'csr')
```

Comment:
Remove spaces around `=`.

I'm not sure we want to force CSR. Instead use `p[0].format`